File: data-pipeline/redis-updater/src/main/java/org/ekstep/ep/samza/domain/Event.java
package org.ekstep.ep.samza.domain;


import com.google.gson.Gson;
import org.ekstep.ep.samza.reader.NullableValue;
import org.ekstep.ep.samza.reader.Telemetry;

import java.util.ArrayList;
import java.util.Map;

public class Event {
    private final Telemetry telemetry;

    public Event(Map<String, Object> map) {
        this.telemetry = new Telemetry(map);
    }

    public Map<String, Object> getMap() {
        return telemetry.getMap();
    }

    public String getJson() {
        Gson gson = new Gson();
        String json = gson.toJson(getMap());
        return json;
    }

    public String userId() {
        NullableValue<String> userId = telemetry.read("actor.id");
        return userId.value();
    }

    public String objectUserId() {
        NullableValue<String> objectType = telemetry.read("object.type");
        if (null != objectType.value() && objectType.value().equalsIgnoreCase("User")) {
            NullableValue<String> objectUserId = telemetry.read("object.id");
            return objectUserId.value();
        }
        return null;
    }

    public String getUserSignInType() {
        NullableValue<ArrayList<Map<String, String>>> cdata = telemetry.read("context.cdata");
        ArrayList<Map<String, String>> cdataList = cdata.value();
        if (null != cdataList && !cdataList.isEmpty()) {
            for (Map<String, String> cdataMap : cdataList) {
                if (cdataMap.containsKey("type") && cdataMap.get("type").equalsIgnoreCase("SignupType"))
                    return cdataMap.get("id").toString();
            }

        }

        return null;
    }

    public String getUserLoginType() {
        NullableValue<ArrayList<Map<String, String>>> cdata = telemetry.read("context.cdata");
        ArrayList<Map<String, String>> cdataList = cdata.value();
        if (null != cdataList && !cdataList.isEmpty()) {
            for (Map<String, String> cdataMap : cdataList) {
                if (cdataMap.containsKey("type") && cdataMap.get("type").equalsIgnoreCase("UserRole"))
                    return cdataMap.get("id").toString();
            }
        }
        return null;
    }

    public ArrayList<String> getUserMetdataUpdatedList() {
        NullableValue<String> edata_state = telemetry.read("edata.state");
        if(edata_state.value().equalsIgnoreCase("Update")) {
            NullableValue<ArrayList<String>> edata_props = telemetry.read("edata.props");
            if(null != edata_props) {
                ArrayList<String> edata_props_list = edata_props.value();
                if(null != edata_props_list && !edata_props_list.isEmpty()) {
                    return edata_props_list;
                } else return null;
            } else return null;
        } else return null;
    }


    @Override
    public String toString() {
        return "Event{" +
                "telemetry=" + telemetry +
                '}';
    }


}

File: data-pipeline-flink/user-cache-updater/src/main/scala/org/sunbird/dp/usercache/domain/Event.scala
package org.sunbird.dp.usercache.domain

import java.util

import org.sunbird.dp.core.domain.Events

class Event(eventMap: util.Map[String, Any]) extends Events(eventMap) {

  override def kafkaKey(): String = {
    did()
  }

  def getId: String = {
    Option(objectType()).map({ t => if (t.equalsIgnoreCase("User")) objectID() else null
    }).getOrElse(null)
  }

  def getState: String = {
    telemetry.read[String]("edata.state").getOrElse(null)
  }

  def getUserSignInType(cDataType: String): String = {
    val cdata = telemetry.read[util.ArrayList[util.Map[String, AnyRef]]]("context.cdata").getOrElse(null)
    var singType: String = null
    if (null != cdata && !cdata.isEmpty) {
      cdata.forEach(cdataMap => {
        if (cdataMap.get("type").asInstanceOf[String].equalsIgnoreCase(cDataType)) singType = cdataMap.get("id").toString else singType
      })
    }
    singType
  }

  def userMetaData():util.ArrayList[String] = {
    telemetry.read[util.ArrayList[String]]("edata.props").getOrElse(null)
  }

}
